File: solutions/go/lasagna/1/lasagna.go
package lasagna

// TODO: define the 'OvenTime' constant
const OvenTime = 40
const preparationMinutesPerLayer = 2

// RemainingOvenTime returns the remaining minutes based on the
// `actual` minutes already in the oven.
func RemainingOvenTime(actualMinutesInOven int) int {
	return OvenTime - actualMinutesInOven
}

// PreparationTime calculates the time needed to prepare the lasagna
// based on the amount of layers.
func PreparationTime(numberOfLayers int) int {
	return numberOfLayers * preparationMinutesPerLayer
}

// ElapsedTime calculates the total time needed to create and bake a
// lasagna.
func ElapsedTime(numberOfLayers, actualMinutesInOven int) int {
	// panic("ElapsedTime not implemented")
	var remainingTime int = RemainingOvenTime(actualMinutesInOven)
	var perLayerTime  int = PreparationTime(numberOfLayers)
	return OvenTime - remainingTime + perLayerTime
}

File: solutions/lua/lasagna/1/lasagna.lua
local lasagna = {
    oven_time         = 40,
    minutes_per_layer =  2,
}

function lasagna.remaining_oven_time(actual_minutes_in_oven)
    return lasagna.oven_time - actual_minutes_in_oven
end

function lasagna.preparation_time(number_of_layers)
    return lasagna.minutes_per_layer * number_of_layers
end

function lasagna.elapsed_time(number_of_layers, actual_minutes_in_oven)
    return lasagna.minutes_per_layer * number_of_layers + actual_minutes_in_oven
end

return lasagna
